### ISSUE TITLE
[IOS][WD Proxy] Got an unexpected response with status undefined: {"code":"ECONNRESET"} Original error: Could not proxy command to remote server.Error: socket hang up

### ISSUE THREAD

**Issue by AymarN** [appium/appium#13387]
## The problem

I'm working on an IOS App with MacOSCatalina 10.15. The WebAgent project is built successfully and WebAgentRunner-Runner.App is reachable by Appium Server.
There are similar issues related to this error however I'm not working on Android so it is no use uninstalling io.appium.uiautomator2 which is not listed in my package.
@KazuCocoa  Below the logs you were asking for. Can you take a look ?
@mykola-mokhnach  why did you close #13320 . The logs were not sent !

## Environment

Appium version (or git revision) that exhibits the issue: 1.15.0
Last Appium version that did not exhibit the issue (if applicable):
Desktop OS/version used to run Appium: macOS Catalina 10.15
Node.js version (unless using Appium.app|exe): v.9.3.0
Npm or Yarn package manager: nom version 5.6.0
Mobile platform/version under test: iOS 12.1
Real device or emulator/simulator: iPhone 6
Appium CLI or Appium.app|exe: CLI

## Details

Xcode 10.2 is used. The Xcode project is inside my workspace and the bundle id created in .xcodeproject is pasted to the app in .properties
 WebDriverAgentRunner-Runner.App  is targeted in Xcodeproject and the bundleId was updated

## Link to Appium logs

https://gist.github.com/AymarN/4ed027abf0a53564805f5fbab478a703
Below Appium logs, iOS Logs and Xcode Logs

## Code To Reproduce Issue [ Good To Have ]

Please remember that with sample code it's easier to reproduce the bug and it's much faster to fix it.

Please git clone https://github.com/appium/appium and from the `sample-code` directory, use one of your favourite languages and sample apps to reproduce the issue.

In case a similar scenario is missing in sample-code, please submit a PR with one of the sample apps provided.

**Comment by mykola-mokhnach** [comment]
> [debug] [WD Proxy] Proxying [POST /session] to [POST http://localhost:8100/session] with body: {"capabilities":{"firstMatch":[{"bundleId":"fr.Bouyguestelecom.WebDriverAgent","arguments":[],"environment":{},"eventloopIdleDelaySec":0,"shouldWaitForQuiescence":true,"shouldUseTestManagerForVisibilityDetection":false,"maxTypingFrequency":60,"shouldUseSingletonTestManager":true}],"alwaysMatch":{}}}
> [IOSDeviceLog] Oct  8 12:39:12 iPhone WebDriverAgentRunner-Runner(XCTest)[1180] <Error>: Exiting due to IDE disconnection.
> [IOSDeviceLog] Oct  8 12:39:12 iPhone backboardd(IOKit)[56] <Notice>: Connection removed: IOHIDEventSystemConnection uuid:70E8637C-A9FB-4982-84AE-1EFD768FBF40 pid:1180 process:WebDriverAgentRunner-Runner type:Passive entitlements:0x0 caller:BackBoardServices: <redacted> + 364 attributes:<CFBasicHash 0x101930f30 [0x101800010]>{type = immutable dict, count = 3,

**Comment by AymarN** [comment]
Thanks for your reply.
I have another issue:
 `[31morg.openqa.selenium.UnsupportedCommandException:` Method is not implemented
 Build info: version: '3.7.1', revision: '8a0099a', time: '2017-11-06T21:01:39.354Z'`

And below the line which is in error:
`driver.getPageSource().contains(...)`
I'm using Appium beta:
1.16.0-beta.1
And I have an additional question. which Appium client version should I use with this server ?

**Comment by reply2sagar** [comment]
I am also seeing this issue. What's the solution. Tests are failing randomly at any point due to this. Do not why  socket hangs up. Do we need to stop the appium server after each test?

**Comment by AymarN** [comment]
@reply2sagar, sorry I would like to know if you use the executor cause in your case your tests are run randomly.
`AppiumCommandExecutor executor = new AppiumCommandExecutor(MobileCommand.commandRepository, new URL("http://localhost:4723/wd/hub"), clientFactory);`
If so, let me know how you implement it. I'm working with Appium client 6.0.0-BETA2

**Comment by AymarN** [comment]
@mykola-mokhnach @KazuCocoa 
I update my Appium client to 6.0.0-BETA2
I try implementing the executor :
Below my import : 
`import org.openqa.selenium.remote.internal.ApacheHttpClient;
import org.openqa.selenium.remote.internal.ApacheHttpClient.Factory;
import org.openqa.selenium.remote.internal.HttpClientFactory;
import io.appium.java_client.remote.AppiumCommandExecutor;`

And how I instantiate the executor : 
`Factory ClientFactory = new ApacheHttpClient.Factory(new HttpClientFactory(60000, 60000));`

I'm trying to get something like this:
`AppiumCommandExecutor executor = new AppiumCommandExecutor(MobileCommand.commandRepository, new URL("http://localhost:4723/wd/hub"), clientFactory); `

Should I use the executor command before my capabilities ?
Do you have snippets or examples to provide  and how to use it.
Thank you for your attention in this matter.

**Comment by viktorgogulenko** [comment]
I was seeing the same issue when I was trying to run tests at the same time on different devices in parallel. It was fixed by using specific port capability for each device:
for iOS devices (range could be used from 8100 till 8199):
> 'appium:wdaLocalPort': 8100

for Android devices (range could be used from 8200 till 8299):
> appium:systemPort': 8200

So each device should have its own port. As soon as each device received its own port in capabilities, issue was resolved and not seeing it even with 10 devices in parallel.

**Comment by mykola-mokhnach** [comment]
Kindly use Appium forum to ask questions.
Closed as not a server issue

**Comment by AymarN** [comment]
@viktorgogulenko I will try your solution . by the way did you use the executor ?
How import did you use ? And how did you implement it ?

AppiumCommandExecutor executor = new AppiumCommandExecutor(MobileCommand.commandRepository, new URL("http://localhost:4723/wd/hub"), clientFactory);
